.github/workflows/test_e2e_sepolia.yaml

```diff
@@ -56,31 +60,43 @@ jobs:
               - '.github/workflows/test_e2e_bonsai_opsepolia.yaml'
               - '.github/actions/test-e2e/**'
 
+  # The gas price on the Ethereum Sepolia network can vary significantly.
+  # To prevent running tests when gas prices are prohibitively high, this job checks the current gas price.
+  # If the gas price exceeds a predefined threshold, the test is skipped to avoid running out of test eth.
   check-gas:
-    name: Check Gas Price
+    name: Check gas price
     runs-on: ubuntu-latest
     outputs:
       gas-ok: ${{ steps.set-result.outputs.gas-ok }}
     steps:
-      - name: Download gas_price_checker from S3
-        uses: keithweaver/aws-s3-github-action@v1.0.0
+      - name: Configure AWS creds
+        uses: aws-actions/configure-aws-credentials@v4
         with:
-          command: cp
-          source: "s3://${{ vars.AWS_BUCKET }}/gas_price_checker"
-          destination: ./gas_price_checker
-          aws_access_key_id: ${{ vars.AWS_ACCESS_KEY_ID }}
-          aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-          aws_region: eu-north-1
-
-      - name: Run gas price checker
+          aws-access-key-id: ${{ vars.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: eu-north-1
+
+      - name: Download gas_price_checker from S3
+        run: |
+          aws s3 cp \
+            s3://${{ vars.AWS_BUCKET }}/gas_price_checker \
+            "$RUNNER_TEMP/gas_price_checker"
+
+      - name: Make gas_price_checker executable
+        run: sudo chmod +x "$RUNNER_TEMP/gas_price_checker"
+
+      - name: Execute gas price checker
         id: check
         continue-on-error: true
```

Comment:
Let's remove `continue-on-error` (which made us overlook the broken job here), and handle it differently.